### ISSUE TITLE
Literate programming for agda

### ISSUE THREAD

**Issue by jonaprieto** [getnikola/nikola#3296]
Dear Nikola devs,

I would like to change my current project to use Nikola, It seems
possible but I don't have too much time nowadays. So, can I get some
help to support literate [Agda](http://agda.readthedocs.io) files?

I would like to replace my Jekyll blog: https://jonaprieto.github.io/mini-hott .
It takes quite a lot to build all the pages and the build is a tricky process.
Nikola seems pretty fast and easy to customise.

So, the source of this blog is at https://github.com/jonaprieto/mini-hott.
The point is that the workflow (using my makefile) is  a bit long.

- The post are in a folder 'src'.
- Each post contains literate agda (\begin{code}...\end{code}) and they can contain Latex code as well. I see there is a Nikola that can do this Latex job.  These files are also markdown files. 
So in summary: literate agda files (.lagda, or .lagda.tex) := Markdown + Latex + Agda.
In addition,
- This files contains unicode. 
- To get HTML from a file .agda, the Agda compiler has an option (--html) or latex (--latex).
- I process that output con Jekyll to get the final connect. 
... I can continue...

Anyone pro to support this?

**Comment by ralsina** [comment]
I would do it in a plug-in, not in the main codebase

Never heard of these, but we have a ton of different markups supported.
It's probably doable.

**Comment by ralsina** [comment]
If there is a command that produces HTML, then Nikola can probably support it.

Something like the kramdown plugin could work as a base.

https://github.com/getnikola/plugins/blob/master/v7/kramdown/

**Comment by jonaprieto** [comment]
No longer interested in using Nikola, maybe later. Thanks anyway for the time.
